### ISSUE TITLE
mock_dynamodb2 mocked table range key is not working

### ISSUE THREAD

**Issue by Yan0417** [spulec/moto#3565]
I found that the range query for a mocked table is sort key agnostic. Meaning the range query on the mocked table will still work even if the given sort key name is wrong.

Here is an example to reproduce this issue:
```
from moto import mock_dynamodb2, mock_s3

with mock_dynamodb2():
    ddb_client = boto3.client("dynamodb", region_name="us-east-1")
    ddb_resource = boto3.resource("dynamodb", region_name="us-east-1")

    # Create mock table
    table = ddb_client.create_table(
        TableName="event",
        KeySchema=[
            {"AttributeName": "uid", "KeyType": "HASH"},
            {"AttributeName": "time", "KeyType": "RANGE"},
        ],
        AttributeDefinitions=[
            {"AttributeName": "uid", "AttributeType": "S"},
            {"AttributeName": "time", "AttributeType": "S"},
        ],
        ProvisionedThroughput={
            'ReadCapacityUnits': 1,
            'WriteCapacityUnits': 1
        }
    )
    ddb_client.get_waiter("table_exists").wait(TableName="event")

    # Insert dummy record
    ddb_client.put_item(
        TableName='event',
        Item={
            'uid': {'S': "user1"},
            'time': {'S': '2020-11-30T06:00:45.823224'},
            'type': {'S': 'normal'},
    })

    ddb_client.put_item(
        TableName='event',
        Item={
            'uid': {'S': "user1"},
            'time': {'S': '2020-11-30T06:00:46.290062'},
            'type': {'S': 'normal'},
        })

    ddb_client.put_item(
        TableName='event',
        Item={
            'uid': {'S': "user1"},
            'time': {'S': '2020-11-30T20:42:04.311627'},
            'type': {'S': 'normal'},
        })

    response = ddb_client.query(
        TableName="event",
        KeyConditionExpression='uid = :a AND not_time BETWEEN :start AND :end',
        ExpressionAttributeValues={
            ':a': {'S': 'user1'},
            ':start': {'S': '2020-11-30T06:00:45.823224'},
            ':end': {'S': '2020-11-30T06:00:46.290062'}
        }
    )

    assert len(response["Items"]) == 2
```

In short, the above example created event table with sort key as `time`, however, when query the table with sort key=`not_time`, I'm still able to fetch the records in the range. This is not expected.

But if I use the similar code to query a real table, I should get below exception:
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the Query operation: Query condition missed key schema element: `time`

FYI:
boto3 version: '1.16.19'
moto version: '1.3.13'

**Comment by bblommers** [comment]
Thanks for raising this @Yan0417 - that doesn't seem right. 
Marking it as an enhancement to implement this validation.

**Comment by Yan0417** [comment]
Thanks for your attention, looking forward to the updates.

**Comment by bblommers** [comment]
I believe this was fixed in #4379. As of moto >= 2.2.9.dev27, the above example will throw the same exception as AWS. (`Query condition missed key schema element: time`)

I'll close this, but let us know if we've missed anything.
